microprofile-jwt/src/test/java/org/jboss/eap/qe/microprofile/jwt/security/keyproperties/KeySizeTestCase.java

```diff
@@ -0,0 +1,158 @@
+     * key 2048 bits long. Proper public key is supplied to server.
+     * @tpPassCrit Authentication is successful and client receives raw token value in response.
+     * @tpSince EAP 7.4.0.CD19
+     */
+    @Test
+    @OperateOnDeployment(BITS_2048_KEY_DEPLOYMENT)
+    public void testJwtSignedBy2048bitsKey(@ArquillianResource URL url) throws URISyntaxException {
+        final RsaKeyTool rsaKeyTool = RsaKeyTool.newKeyTool(getPrivateKey("key.private.pkcs8.pem"));
+
+        JsonWebToken token = new JwtHelper(rsaKeyTool, "issuer").generateProperSignedJwt();
+
+        RestAssured.given().header("Authorization", "Bearer " + token.getRawValue())
+                .when().get(url.toExternalForm() + "basic-endpoint")
+                .then().body(equalTo(token.getRawValue()));
+    }
+
+    /**
+     * @tpTestDetails Test specification compatibility by authenticating against server with a token signed with a valid
+     * key 2048 bits long. Proper public key is supplied to server.
+     * @tpPassCrit Authentication is successful and client receives raw token value in response.
+     * @tpSince EAP 7.4.0.CD19
+     */
+    @Test
+    @OperateOnDeployment(BITS_4096_KEY_DEPLOYMENT)
+    public void testJwtSignedBy4096bitsKey(@ArquillianResource URL url) throws URISyntaxException {
+        final RsaKeyTool rsaKeyTool = RsaKeyTool.newKeyTool(getPrivateKey("key4096.private.pkcs8.pem"));
+
+        JsonWebToken token = new JwtHelper(rsaKeyTool, "issuer").generateProperSignedJwt();
+
+        RestAssured.given().header("Authorization", "Bearer " + token.getRawValue())
+                .when().get(url.toExternalForm() + "basic-endpoint")
+                .then().body(equalTo(token.getRawValue()));
+    }
+
+    /**
+     * @tpTestDetails Test specification compatibility by authenticating against server with a token signed with a valid
+     * key 512 bits long. This is a negative test scenario and should not succeed since MP-JWT 1.1 supports only 1024
+     * and 2048 bits long keys. Proper public key is supplied to server.
+     * @tpPassCrit Authentication is not successful and client receives "unauthorized" response.
+     * @tpSince EAP 7.4.0.CD19
+     */
+    @Test
+    @OperateOnDeployment(BITS_512_KEY_DEPLOYMENT)
+    public void testJwtSignedBy512bitsKey(@ArquillianResource URL url) throws URISyntaxException {
+        final RsaKeyTool rsaKeyTool = RsaKeyTool.newKeyTool(getPrivateKey("key512.private.pkcs8.pem"));
+
+        final JsonWebToken token = new JwtHelper(rsaKeyTool, "issuer").generateProperSignedJwt();
+
+        RestAssured.given()
+                .header("Authorization", "Bearer " + token.getRawValue())
+                .when().get(url.toExternalForm() + "basic-endpoint")
+                .then()
+                .body(equalTo("<html><head><title>Error</title></head><body>Unauthorized</body></html>"))
+                .and()
+                .statusCode(401);
+
+        //TODO add check for expected log message with correct reason
```

Comment:
What is the `TODO` related to?

Reply:
Thanks, removed for now, since the situation about JWT logs is a little bit unclear for now...